### PR TITLE
Add nanosecond precision to :interval: calc

### PR DESCRIPTION
**Pull request by timraymond**
The UI for selecting time ranges produces nanosecond-precision times
for the template variables in the absolute time range. The regex for
absolute time ranges previously did not account for this precision,
which caused it to break.

This updates the regex to consider the nanosecond parts of an
RFC3339Nano timestamp and updates the associated tests. The Gobular link
has also been updated to a fresh playground used to debug this issue.

  - [ ] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [x] Rebased/mergable
  - [x] Tests pass

Connect #1905

### CODE REVIEW

File: chronograf.go [hunk 1]
```diff
@@ -260,7 +260,7 @@ func (g *GroupByVar) parseRelative(fragment string) (time.Duration, error) {
 // example, the fragement "time > '1985-10-25T00:01:21-0800 and time <
 // '1985-10-25T00:01:22-0800'" would yield a duration of 1m'
 func (g *GroupByVar) parseAbsolute(fragment string) (time.Duration, error) {
-	timePtn := `time\s[>|<]\s'([0-9\-TZ\:]+)'` // Playground: http://gobular.com/x/41a45095-c384-46ea-b73c-54ef91ab93af
+	timePtn := `time\s[>|<]\s'([0-9\-T\:\.Z]+)'` // Playground: http://gobular.com/x/208f66bd-1889-4269-ab47-1efdfeeb63f0
```
Comment (goller) [inline chronograf.go]: Would you also reverse the statement?

`'([0-9\-T\:\.Z]+)'\s[>|<]\stime`

Comment (goller) [inline chronograf.go]: Also, documentation claims that time literals can have a space.  I don't know if this is true.  Would you find out if this is true?

https://docs.influxdata.com/influxdb/v1.3/query_language/spec/#literals

time_lit            = "2006-01-02 15:04:05.999999" | "2006-01-02" .

Comment (timraymond) [reply chronograf.go]: @goller The reversed regex does not parse the expression: http://gobular.com/x/dac04b30-e4ba-466c-b7b9-44966f059e10

Comment (goller) [reply chronograf.go]: @timraymond I'm suggesting this : http://gobular.com/x/d5cf2d64-1361-469b-90db-fdf79583bf48

**Comment by 121watts** [comment]
connect #1925

**Comment by timraymond** [comment]
@goller and I discussed last week, and the remaining feedback will be addressed in another PR. Merging.

**Pull request merged**